### ISSUE TITLE
New Rule request: spacing-default-value

### ISSUE THREAD

**Issue by sc0ttyd** [eslint/eslint#7035]
AFAICT there is no rule that covers spacing for default values in function arguments or destructuring. I would like to be able to enforce this:

```js
// bad
const { id, name='no name' } = user;

// good
const { id, name = 'no name' } = user;

// bad
function(text, style='none') { //... }

// good
function(text, style = 'none') { //.. }
```

This is obviously a stylistic rule. This warrants inclusion since in almost all other situations in which spacing preference can be expressed there is a corresponding core rule. As discussed in #3299 the equals in this context does not function as an infix operator, so the existing `space-infix-ops` rule does not apply. Maybe it might be best to consider this as part of the remit of `space-infix-ops` rather than creating a new rule.

**Comment by eslintbot** [comment]
Hi @sc0ttyd, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
Related: There's a discussion about augmenting space-infix-ops to differentiate between assignment and default parameters in issue #3587.

**Comment by vitorbal** [comment]
Seems like this is already planned in #3587 as @platinumazure mentions, so closing this in favor of that issue. Thanks for reaching out, @sc0ttyd!
